File: Sorting Algorithms/InsertionSort.java
public class InsertionSortDemo {

    public void insertionsort(int A[], int n) {
        for(int i=1;i<n;i++) {
            int temp = A[i];
            int position = i;
            while(position>0 && A[position-1] > temp) {
                A[position] = A[position-1];
                position = position - 1;
            }
            A[position] = temp;
        }
    }

    public void display(int A[], int n) {
        for(int i=0;i<n;i++)
            System.out.print(A[i] + " ");
        System.out.println();
    }

    public static void main(String args[]) {
        InsertionSortDemo s = new InsertionSortDemo();
        int A[] = {3, 5, 8, 9, 6, 2};
        System.out.println("Original Array: ");
        s.display(A, 6);
        s.insertionsort(A, A.length);
        System.out.println("Sorted Array: ");
        s.display(A, 6);
    }

}

File: Sorting Algorithms/Python/insertion_sort.py
def insertion_sort(arr):
    """
    Insertion Sort Algorithm
    
    Parameters:
    - arr: List to be sorted
    """
    # Traverse through all array elements
    for i in range(1, len(arr)):
        key = arr[i]
        
        # Move elements of arr[0..i-1] that are greater than key to one position ahead of their current position
        j = i-1
        while j >=0 and key < arr[j] :
                arr[j+1] = arr[j]
                j -= 1
        arr[j+1] = key

# Example Usage:
arr = [12, 11, 13, 5, 6]
insertion_sort(arr)
print("Insertion Sorted array:", arr)
